NitroxClient/Communication/Packets/Processors/FireDousedProcessor.cs

```diff
@@ -22,11 +25,37 @@ public FireDousedProcessor(IPacketSender packetSender)
         /// </summary>
         public override void Process(FireDoused packet)
         {
-            GameObject fireGameObject = NitroxEntity.RequireObjectFrom(packet.Id);
+            Optional<GameObject> gameObject = NitroxEntity.GetObjectFrom(packet.Id);
+            if (!gameObject.HasValue)
+            {
+                // cannot find that ID. try to douse something at the same location.
```

Comment:
Is it possible that we can deterministically know these IDs ahead of time.  For example, if there are N number of fire spawn points, we can make them deterministic based on parent cyclops id (NitroxId.nextId()).

Reply:
That goes way beyond my knowledge of Unity, Nitrox or Subnautica :)
I'm just glad the ParentID was known in case of Aurora ... will try to find out how to start fires on Cyclops this evening and see if my patch handles those well.

Are there other places where a Fire might show up?

Reply:
Sorry for the slow reply, it is the holidays.

After examining the code, it looks like cyclops fires will always lookup a valid gameobject due to being tagged with an id on spawn.  So you don't need to be concerned with that use-case.

However, I think this implementation may only work for local players that are in the immediate area.  If one player extinguishes the fire, while the other one is far away, then the other player will still see the fire.  Can you verify that this is the case?

I think the right long-term fix is for the server to know about the fires so that it can appropriately allocate an id on spawn.  That would allow you to use Nitrox's EntityMetadata system to store the douse amount; that way, when players enter the area, it will automatically apply the correct douse and even persist it on server reset.  

Example metadata: https://github.com/SubnauticaNitrox/Nitrox/blob/master/NitroxClient/GameLogic/Spawning/Metadata/KeypadMetadataProcessor.cs

Area where we spawn (see if the parent is there by adding breakpoint on class id): https://github.com/SubnauticaNitrox/Nitrox/blob/8815fb0a8fc80ff030bc12c46dce10fe380febde/NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs#L265

Reply:
Just to update this.  The server knows about the parent objects in the aurora.  Thus, if the fire does not have a nitrox id, you can go one level up to get it.  Then you can use the entity metadata system to persist this data.  That will remove the need for the location checks and will work for out of range players.